Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -101,6 +107,109 @@ public virtual Fill Fill(FillModelParameters parameters)
+            var fill = new OrderEvent(order, utcTime, OrderFee.Zero);
+
+            //If its cancelled don't need anymore checks:
+            if (order.Status == OrderStatus.Canceled) return fill;
+
+            // Fill only if open or extended
+            if (!IsExchangeOpen(asset,
+                Parameters.ConfigProvider
+                    .GetSubscriptionDataConfigs(asset.Symbol)
+                    .IsExtendedMarketHours()))
+            {
+                return fill;
+            }
+
+            // Get the range of prices in the last bar:
+            var tradeHigh = 0m;
+            var tradeLow = 0m;
+            var endTimeUtc = DateTime.MinValue;
+
+            var subscribedTypes = GetSubscribedTypes(asset);
+
+            if (subscribedTypes.Contains(typeof(Tick)))
+            {
+                var trade = asset.Cache.GetAll<Tick>().LastOrDefault(x => x.TickType == TickType.Trade && x.Price > 0);
+                if (trade != null)
+                {
+                    tradeHigh = trade.Price;
+                    tradeLow = trade.Price;
+                    endTimeUtc = trade.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            else if (subscribedTypes.Contains(typeof(TradeBar)))
+            {
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+
+                if (tradeBar != null)
+                {
+                    tradeHigh = tradeBar.High;
+                    tradeLow = tradeBar.Low;
+                    endTimeUtc = tradeBar.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+
+            // do not fill on stale data
+            if (endTimeUtc <= order.Time) return fill;
+
+            //Check if the limit if touched order was filled:
+            switch (order.Direction)
+            {
+                case OrderDirection.Buy:
+                    //-> 1.2 Buy: If Price below Trigger, Buy:
+                    if (tradeLow <= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+                        var askCurrent = GetAskPrice(asset, out endTimeUtc);
+
+                        if (askCurrent <= order.LimitPrice)
+                        {
+                            fill.Status = OrderStatus.Filled;
+                            fill.FillPrice = Math.Min(askCurrent, order.LimitPrice);
+                            fill.FillQuantity = order.Quantity;
+                        }
+                    }
+
+                    break;
+
+                case OrderDirection.Sell:
+                    //-> 1.2 Sell: If Price above Trigger, Sell:
+                    if (tradeHigh >= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+                        var bidCurrent = GetBidPrice(asset, out endTimeUtc);
+
+                        if (bidCurrent >= order.LimitPrice)
```

Comment:
Here we can't know the order of things happening in a bar: if the LIT order was triggered, using trades, before the quote low, so we use **current** quote information instead. In practice, it might leave trades out of happening. This is following the existing behavior of `StopLimitFill`. Let's please add this as a comment in the description of this method and the `FillModel` version too 👍